htsinfer/rand_nuc_seq.py

```diff
@@ -0,0 +1,249 @@
+from collections import Counter
+
+import numpy as np
+
+def randomize_nucleotide_sequence(input_sequences: list,
```

Comment:
Type hints with literal `list` and `dict` are only really supported since Python 3.9, which is only a couple of weeks old. For backwards compatibility it is better to use `List` and `Dict` here, which you need to import from the `typing` package.

Below the import from `collections`, add:

```py
from typing import list
```

Then replace the type hints for `list` with `List`.